super-csv/src/test/java/org/supercsv/io/TokenizerTest.java

```diff
@@ -586,6 +587,178 @@ public void testQuotedSpaces() throws Exception {
 		assertEquals("   three   ", columns.get(2));
 		assertEquals(input, tokenizer.getUntokenizedRow());
 	}
+
+	/**
+	 * Tests the readColumns() method with a quote character escaped with another quote (RFC style)
+	 */
+	@Test
+	public void testQuotedQuoteCharWithDoubleQuoteEscape() throws Exception {
+		final String input = "one,\"two\",\"field with \"\" quote char\",four";
+		tokenizer = createTokenizer(input, NORMAL_PREFERENCE);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 4);
+		assertEquals("one", columns.get(0));
+		assertEquals("two", columns.get(1));
+		assertEquals("field with \" quote char", columns.get(2));
+		assertEquals("four", columns.get(3));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with a quote character escaped with another quote (RFC style)
+	 */
+	@Test
+	public void testQuotedQuoteCharWithBackslashEscape() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "one,two,\"field with \\\" quote char\",four";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 4);
+		assertEquals("one", columns.get(0));
+		assertEquals("two", columns.get(1));
+		assertEquals("field with \" quote char", columns.get(2));
+		assertEquals("four", columns.get(3));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an escape character following another escape char.
+	 */
+	@Test
+	public void testEscapedEscapeChar() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"field with \\\\ escape char\"";
+		tokenizer = createTokenizer(input, csvPref);
+		tokenizer.readColumns(columns);
+
+		assertTrue(columns.size() == 1);
+		assertEquals("field with \\ escape char", columns.get(0));
+		assertEquals(input, tokenizer.getUntokenizedRow());
+	}
+
+	/**
+	 * Tests the readColumns() method with an even number of escape characters
+	 * char.
+	 */
+	@Test
+	public void testEvenSeriesOfEscapeChars() throws Exception {
+
+		final CsvPreference csvPref = new CsvPreference.Builder(NORMAL_PREFERENCE)
+				.setQuoteEscapeChar('\\')
+				.build();
+
+		final String input = "\"\\\\\\\\\\\\\"";
```

Comment:
It's tricky because I also have to escape backslashes to write out a String literal in Java..

The implementation I've created interprets a doubled escape char inside a _quoted_ field as just one escape char.  But a doubled escape char in an _unquoted_ field as two literal escape chars.  So this covers the case of 1) quoted field 2) doubled escape char.  The other one (odd series of escape chars) covers 1) quoted field 2) doubled escape char with a trailing escape char.  That should fail (and does).

I've added another test for the 1) unquoted field and 2) doubled escape char called `testDoubledEscapeCharInUnquotedFieldIsLiteral` that I'm about to push also.